### PR TITLE
#37 Spatial Partitioning

### PR DESCRIPTION
**Pull request by zoompower**

### CODE REVIEW

File: Hivemind/Assets/Scripts/SpatialPartitioning.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections;
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scripts/SpatialPartitioning.cs]: Remove unused usings plz

File: Hivemind/Assets/Scripts/SpatialPartitioning.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        //TO DO, delete this when done DEBUGGING
+        if(Entities.Count != 0)
+        {
+            Debug.ClearDeveloperConsole();
+            Debug.Log(this.name+" Count: "+Entities.Count);
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scripts/SpatialPartitioning.cs:28]: Please comment out or delete the hardcore spamming logger

File: Hivemind/Assets/Scripts/SpatialPartitioning.cs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    }
+
+    private void OnTriggerExit(Collider entity)
+    {
+        Entities.Remove(entity.gameObject);
+    }
+
+    public List<GameObject> GetEntities()
+    {
+        return Entities;
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scripts/SpatialPartitioning.cs]: No need to get a property which is public, delete getter or make entities private

File: Hivemind/Assets/Scripts/SpatialPartitioning.cs [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    }
+
+    public List<GameObject> GetEntitiesWithNeigbors()
+    {
+        List<GameObject> EntitiesWithNeighbors = GetEntities();
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scripts/SpatialPartitioning.cs]: Why use the Getter when you have the property in your class?

File: Hivemind/Assets/Scripts/SpatialPartitioning.cs [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+            {
+                EntitiesWithNeighbors.Add(ant);
+            }
+        }
+        return EntitiesWithNeighbors;
+    }
+
+    public List<GameObject> GetEntitiesWithExtraNeighbors(int steps)
+    {
+        List<GameObject> EntitiesWithExtraNeighbors = GetEntities();
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scripts/SpatialPartitioning.cs]: Why use the Getter when you have the property in your class?

File: Hivemind/Assets/Scripts/SpatialPartitioning.cs [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+        for (int i = -steps; i<steps;i++) {
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scripts/SpatialPartitioning.cs]: { on next line plz

**Review by ErikLouwerse — APPROVED** [review]

**Review by MMCQ33 — APPROVED** [review]

**Pull request merged**